### PR TITLE
[references] Add poly scheduler for detection training

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- Add Polynomial Scheduler for detection training (TF & PT)
- It's often mentioned in text detection papers like DB & FAST & .. so i think it is a good idea to add it :)

Any feedback is welcome :hugs:

### CODE REVIEW

File: references/detection/train_tensorflow.py [hunk 1]
```diff
@@ -268,14 +268,25 @@ def main(args):
...
+        )
+    elif args.sched == "poly":
+        scheduler = tf.keras.optimizers.schedules.PolynomialDecay(
+            args.lr,
+            decay_steps=args.epochs * len(train_loader),
+            end_learning_rate=1e-7,
+            power=1.0,
+            cycle=False,
+            name="PolynomialDecay",
+        )
```
Comment (odulcy-mindee) [inline references/detection/train_tensorflow.py:288]: It should raise an error if `args.sched` is not matched with previous values

**Comment by felixdittrich92** [comment]
@odulcy-mindee I think with choices it's more clean instead of raising an exception :)
Switched the default to poly -> (After testing LinkNet / DB & FAST on 2 different detection datasets x Poly / Cosine / Onecycle -> Poly has reached the best results with mostly no drops)

**Review by odulcy-mindee — APPROVED** [review]
Yeah, it's better with choices, thanks !
